### DISCUSSION TITLE
FileLocator （和 Listary）和 Obsidian 联用（需要 FileMenu Tools）

### DISCUSSION THREAD

**Discussion by Quorafind** [obsidianzh/forum#29] [General综合]
这里分享两个非常非常简单的脚本给那些用 filemenu tools 和 filelocator 的人：
```powershell
$DefaultNoteName = Get-Clipboard
$RealNoteName = $DefaultNoteName -replace ' ', '%20'
$RealNoteURL = "obsidian://open?vault=仓库名&file=" + $RealNoteName
Start-Process $RealNoteURL
```
和
```bat
"file menu tools安装目录\App\FileMenu Tools\cmdfmt64.exe" CopyNameToClipboard %1
PowerShell.exe -WindowStyle Hidden -file "脚本位置"
```
这两个脚本的目的是为了能够用 file menu tools 在右键加上一个能直达 ob 对应笔记的选项。然后再在 Obsidian 中加上一个 跳转到 Obsidian 的命令，选用为启动这个 bat脚本，且传入 `"%FILEPATH1%"` 即可。

注意，上述中的**仓库名、安装目录、脚本位置**都是要自己手动填入的

PLUS 版：
在 Listary 的命令中填入以下内容：
> fob 
> Filelocator 
> 用 Filelocator 搜索 OB 的 '{query}'
> D:\software\FileLocator Pro\FileLocatorPro.exe
> -s -r -d "D:\OneDrive\1-学习文档\obsidianStuff\MyBrain;!D:\OneDrive\1-学习文档\obsidianStuff\MyBrain\.trash" -f "*.md" -c "{query}"

如下图所示：
![image](https://user-images.githubusercontent.com/13215013/102158747-16f79f00-3ebd-11eb-9885-54f0b822b56f.png)

即可实现 Listary 搜索指令，filelocator 打开 Ob 的效果了。

**Comment by Quorafind** [comment]
可能遇到的情况：
1、无法启动脚本，解决方法：打开PowerShell 然后输入 `get-executionpolicy` 检查是否设置了严格的限制，如果是的话，则输入 `set-executionpolicy remotesigned` 去除限制。

**Comment by cyddgh** [comment]
我将本文内容进行了转述、简化和演绎，发表在微信平台上 https://mp.weixin.qq.com/s/c5xVxTh-W1wwR3sdqUbg1w
